From aa7bf4c49d40702a54e865391ec694ca211b0e0e Mon Sep 17 00:00:00 2001
From: Artem Iglikov <artikz@google.com>
Date: Thu, 15 Feb 2018 09:12:58 +0000
Subject: [PATCH 134/306] Revert "Use structural equality for
 A11yNodeInfo#mChildNodeIds"

This reverts commit df99184a81c4a9be3073dc3e6d396a950c9c6be2.

Reason for revert: Crashes settings - b/73440274

Bug: 73440274
Change-Id: I529874a6110e04d6b5b7723906a03330cd637d1d
(cherry picked from commit b45e0b099f24d13b6f3aaceaf5cf52f2118ce57f)
---
 core/java/android/util/LongArray.java           | 17 +----------------
 .../accessibility/AccessibilityNodeInfo.java    |  2 +-
 2 files changed, 2 insertions(+), 17 deletions(-)

diff --git a/core/java/android/util/LongArray.java b/core/java/android/util/LongArray.java
index f9454d962eb..9b0489ca5c6 100644
--- a/core/java/android/util/LongArray.java
+++ b/core/java/android/util/LongArray.java
@@ -18,10 +18,8 @@ package android.util;
 
 import com.android.internal.util.ArrayUtils;
 import com.android.internal.util.Preconditions;
-
-import libcore.util.EmptyArray;
-
 import java.util.Arrays;
+import libcore.util.EmptyArray;
 
 /**
  * Implements a growing array of long primitives.
@@ -218,17 +216,4 @@ public class LongArray implements Cloneable {
             throw new ArrayIndexOutOfBoundsException(mSize, index);
         }
     }
-
-    /**
-     * Test if each element of {@code a} equals corresponding element from {@code b}
-     */
-    public static boolean elementsEqual(LongArray a, LongArray b) {
-        if (a.mSize != b.mSize) return false;
-        for (int i = 0; i < a.mSize; i++) {
-            if (a.get(i) != b.get(i)) {
-                return false;
-            }
-        }
-        return true;
-    }
 }
diff --git a/core/java/android/view/accessibility/AccessibilityNodeInfo.java b/core/java/android/view/accessibility/AccessibilityNodeInfo.java
index 5b1dd5c88ca..417a72530b7 100644
--- a/core/java/android/view/accessibility/AccessibilityNodeInfo.java
+++ b/core/java/android/view/accessibility/AccessibilityNodeInfo.java
@@ -3203,7 +3203,7 @@ public class AccessibilityNodeInfo implements Parcelable {
         fieldIndex++;
         if (mConnectionId != DEFAULT.mConnectionId) nonDefaultFields |= bitAt(fieldIndex);
         fieldIndex++;
-        if (!LongArray.elementsEqual(mChildNodeIds, DEFAULT.mChildNodeIds)) {
+        if (!Objects.equals(mChildNodeIds, DEFAULT.mChildNodeIds)) {
             nonDefaultFields |= bitAt(fieldIndex);
         }
         fieldIndex++;
-- 
2.17.1

